flink-clients/src/main/java/org/apache/flink/client/program/Client.java

```diff
@@ -65,31 +73,25 @@
 /**
  * Encapsulates the functionality necessary to submit a program to a remote cluster.
  */
-public class Client {
+public abstract class Client {
```

Comment:
Okay, I see. Maybe we can reduce the number a bit by reducing the savepoint-compatibility methods.
If these methods are only called in one or two locations, it should be fine to just pass a null for the savepoint.